scripts/update-versions.js

```diff
@@ -0,0 +1,160 @@
+        case 'minor': return `${major}.${newVersionNumber}.0`;
+        case 'patch': return `${major}.${minor}.${newVersionNumber}`;
+    }
+
+    return currentFullVersion;
+}
+
+/**
+ * Will return the updated package version based on the dependencies update
+ * @param {string} update - Either `major`, `minor` or `patch`
+ * @param {string} currentFullVersion
+ * @returns {string}
+ */
+function getUpdatedPackageVersion(update, currentFullVersion) {
+    const [major, minor, patch] = currentFullVersion.split('.');
+
+    switch (update) {
+        case 'major': return `${parseInt(major) + 1}.0.0`;
+        case 'minor': return `${major}.${parseInt(minor) + 1}.0`;
+        case 'patch': return `${major}.${minor}.${parseInt(patch) + 1}`;
+    }
+
+    return currentFullVersion;
+}
+
+/**
+ * Will update the dependencies of the module
+ * @param {string} packageJSONPath
+ * @returns {void}
+ */
+function updateDependencies(packageJSONPath, { moduleVersion, name }) {
+    const packageJSON = JSON.parse(fs.readFileSync(packageJSONPath, { encoding: 'utf-8' }));
+    const packageVersion = packageJSON.version;
+    if (!packageJSON.dependencies) return;
+
+    if (!packageJSON.dependencies[name]) return;
+    const dependencyVersion = packageJSON.dependencies[name].replace('^', '');
+
+    const { update, versionUpdate } = getVersionDiff(moduleVersion, dependencyVersion);
+    // We should update the dependency just if it has major update!
+    if (!update || update !== 'major') return;
+    const newDependencyVersion = getNewVersion(update, dependencyVersion, versionUpdate);
+
+    if (newDependencyVersion === dependencyVersion) return;
+
+    console.log(`Updating '${name}' dependency for ${packageJSON.name} from ${dependencyVersion} to ${newDependencyVersion}`);
+    packageJSON.dependencies[name] = `^${newDependencyVersion}`;
+
+    // If major update has been done of some dependency then update the package major versions as well
+    // because the backwards compatability will be broken
+    // Also check if the package version has been updated already to prevent updating it twice.
+    // When package is updated from 1.0.0 to 2.0.0 and then looping it for second time won't update it to 3.0.0
+    if (!updatedPackages[packageJSON.name] && packageJSON.name !== 'components-tests') {
+        packageJSON.version = getUpdatedPackageVersion(update, packageVersion);
+        console.log(`Updating '${packageJSON.name}' version from ${packageVersion} to ${packageJSON.version}`);
+        updatedPackages[packageJSON.name] = packageJSON.version;
+    }
+
+    fs.writeFileSync(packageJSONPath, JSON.stringify(packageJSON, null, 2) + '\n', { encoding: 'utf-8' });
+}
+
+/**
+ * @param {Object} data
+ * @param {string} data.moduleVersion
+ * @param {string} data.name
+*/
+function updateVersionDependents(data) {
+    const components = getComponentDirectories(['slider', 'scrollable-container', 'dropdown']);

```

Comment:
What is the purpose of getting those specific components? This is the third instance I've seen so far.

Reply:
this defines the order of the components because dropdown depends on scrollable-container that depends on slider.
Otherwise it will use lexical order that is dropdown, scrollable-container, slider and it is wrong because it won't update the versions dependents in the correct order.